cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -132,6 +126,20 @@ public void setAuditPublisher(AuditPublisher auditPublisher) {
    * @throws IOException if there is a problem in making an HTTP request to check if the namespace exists.
    */
   public DatasetMeta get(Id.DatasetInstance instance, List<? extends Id> owners) throws Exception {
+    try {
+      return metaCache.get(instance);
+    } catch (ExecutionException e) {
+      if ((e.getCause() instanceof Exception) && (e.getCause() instanceof HttpErrorStatusProvider)) {
```

Comment:
No, because e.getCause() could be a Throwable. And HttpErrorStatusProvider does not extend Exception. You would get "uncaught Throwable".